src/Var.cc

```diff
@@ -420,7 +420,9 @@ StmtPtr add_local(IDPtr id, TypePtr t, InitClass c, ExprPtr init,
 
 	else
 		{
-		current_scope()->AddInit(std::move(id));
+		if ( c != INIT_SKIP )
```

Comment:
could this be done just by checking for `INIT_NONE` and not bothering to create a new initialization class? I forget whether `INIT_NONE` comes with some sort of activity that would make it inappropriate here.

Reply:
For records, containers and vectors, INIT_NONE does initialize local variables
https://github.com/zeek/zeek/blob/6e6a2bee8af39fb03fd9807a30b6ac3500b1464e/src/Stmt.cc#L1782-L1795

Flipping this to INIT_NONE breaks the following.
```
event zeek_init()
   {
   local v: vector of count;
   v += 1;
   print v;
   }
```

Vectors, records, tables may well be used as index/yield types.

(For records it's a bit curious though, because it allows to construct an "empty" record with no fields set).

I'm resolving this one.